Storage/SQL/BrowserDB.swift

```diff
@@ -36,48 +36,43 @@ open class BrowserDB {
     // instance has been initialized (schema is created/updated).
     public func touch() -> Success {
         let deferred = Success()
-        var err: NSError? = nil
-        withConnection(&err) { connection, error -> Void in
-            guard let _ = connection as? ConcreteSQLiteDBConnection else {
-                deferred.fill(Maybe(failure: DatabaseError(description: "Could not establish a database connection")))
-                return
-            }
 
-            deferred.fill(Maybe(success: ()))
+        do {
+            try withConnection { connection -> Void in
+                guard let _ = connection as? ConcreteSQLiteDBConnection else {
+                    throw DatabaseError(description: "Could not establish a database connection")
+                }
+
+                deferred.fill(Maybe(success: ()))
+            }
+        } catch let err as NSError {
+            deferred.fill(Maybe(failure: err))
         }
 
         return deferred
     }
 
-    func withConnection<T>(flags: SwiftData.Flags, err: inout NSError?, callback: @escaping (_ connection: SQLiteDBConnection, _ err: inout NSError?) -> T) -> T {
+    @discardableResult func withConnection<T>(flags: SwiftData.Flags = .readWriteCreate, _ callback: @escaping (_ connection: SQLiteDBConnection) throws -> T) throws -> T {
         var res: T!
-        err = db.withConnection(flags) { connection in
-            // An error may occur if the internet connection is dropped.
-            var err: NSError? = nil
-            res = callback(connection, &err)
-            return err
-        }
-        return res
-    }
 
-    func withConnection<T>(_ err: inout NSError?, callback: @escaping (_ connection: SQLiteDBConnection, _ err: inout NSError?) -> T) -> T {
-        /*
-         * Opening a WAL-using database with a hot journal cannot complete in read-only mode.
```

Comment:
Keep this comment as a block comment on `withConnection`, now that you've inlined the default. Very few people actually know this, and it's crucial to why we default to `.readWriteCreate`!